api/src/main/java/io/druid/indexer/TaskState.java

```diff
@@ -22,8 +22,11 @@
 public enum TaskState
 {
   RUNNING,
+  WAITING,
+  PENDING,
   SUCCESS,
-  FAILED;
+  FAILED,
+  UNKNOWN;
```

Comment:
What does `UNKNOWN` state mean?

Reply:
So instead of passing null for status, I added this `UNKNOWN` state, when we don't know the state of a task.